tensorflow/compiler/xla/mlir_hlo/tests/Dialect/mhlo/mhlo_infer_shape_type_methods.mlir

```diff
@@ -415,4 +415,31 @@ func.func @add_bounds_unranked(
   // CHECK: types0 = tensor<*xf32>
   %1 = "mhlo_test.get_return_types"(%result) : (tensor<*xf32>) -> tensor<*xindex>
   func.return %1 : tensor<*xindex>
-}
\ No newline at end of file
+}
+
+// -----
+
+// CHECK-LABEL: @reduce
+func.func @reduce(%arg0: tensor<?x4xf32>) -> tensor<?xindex> {
+  %0 = mhlo.constant dense<-0.000000e+00> : tensor<f32>
+  %1 = mhlo.reduce(%arg0 init: %0) applies mhlo.add across dimensions = [1] : (tensor<?x4xf32>, tensor<f32>) -> tensor<?xf32>
```

Comment:
Could you make this a variadic reduce to increase coverage? Ideally with a 2-d result that is only partially dynamic.